src/mainloop.rkt

```diff
@@ -171,51 +166,25 @@
   (define vars (program-variables orig-prog))
   (define err&exprs (localize-error (alt-program (^next-alt^)) (*output-repr*)))
 
-  ; duplicate locations (already seen)
-  (^duplocs^ '())
-
   ; high-error locations
-  (let loop ([err&exprs err&exprs] [count 0])
-    (cond
-     [(or (null? err&exprs) (>= count (*localize-expressions-limit*)))
-      (void)]
-     [else
-      (match-define (cons err expr) (car err&exprs))
-      (cond
-       [(and (*use-improve-cache*)          ; if in cache, put into duplocs
-             (patch-table-has-expr? expr))
-        (timeline-push! 'locations (~a expr) (errors-score err) #f)
-        (^duplocs^ (cons expr (^duplocs^)))
-        (loop (cdr err&exprs) (if (*localize-limit-for-new*) count (+ count 1)))]
-       [else                                ; else, add to list
-        (timeline-push! 'locations (~a expr) (errors-score err) #t)
-        (patch-table-add! expr vars #f)
-        (loop (cdr err&exprs) (+ count 1))])]))
+  (^locs^
+    (for/list ([e&e err&exprs] [i (in-range 4)])
```

Comment:
Weird hack—you can do `[(err expr) (in-dict err&exprs)]` instead of what you do here. It saves coming up with a dumb name.